container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1926,6 +2394,77 @@ func (manager *KubernetesManager) HasComputeNodes(ctx context.Context) (bool, er
 	return len(nodes.Items) != 0, nil
 }
 
+func (manager *KubernetesManager) AddLabelsToNode(ctx context.Context, nodeName string, labels map[string]string) error {
```

Comment:
My understanding is that node management is more done on the k8s admin side. If that's correct, as a k8s admin I'd be suspicious of why Kurtosis is changing node labels

Reply:
hmm - in a way Kurtosis is already doing a lot of node management. we need to add node selectors and affinities/tolerations to pods, thus Kurtosis already has/needs to have knowledge of labels on nodes/node hostnames etc. I think if a k8s admin is using kurtosis in their cluster, an expectation is already set Kurtosis is making some changes - we can be explicit about exactly what changes in v4 config docs about `engine-node-name`